processor/src/chiplets/aux_trace/mod.rs

```diff
@@ -245,3 +304,995 @@ impl AuxColumnBuilder<ChipletsVTableUpdate, ChipletsVTableRow, u32> for Chiplets
+    main_trace: &MainTrace,
+    alphas: &[E],
+    i: usize,
+) -> E {
+    let op_label = LINEAR_HASH_LABEL;
+    let addr_nxt = main_trace.addr(i + 1);
+    let first_cycle_row = addr_to_row_index(addr_nxt) % 8 == 0;
+    let transition_label = if first_cycle_row { op_label + 16 } else { op_label + 32 };
+
+    let header =
+        alphas[0] + alphas[1].mul_base(Felt::from(transition_label)) + alphas[2].mul_base(addr_nxt);
+
+    let state = main_trace.decoder_hasher_state(i);
+
+    header + build_value(&alphas[8..16], &state)
+}
+
+/// Builds requests made to the hasher chiplet at the start of a respan block.
+fn build_respan_block_request<E: FieldElement<BaseField = Felt>>(
+    main_trace: &MainTrace,
+    alphas: &[E],
+    i: usize,
+) -> E {
+    let op_label = LINEAR_HASH_LABEL;
+    let addr_nxt = main_trace.addr(i + 1);
+
+    let first_cycle_row = addr_to_row_index(addr_nxt - ONE) % 8 == 0;
+    let transition_label = if first_cycle_row { op_label + 16 } else { op_label + 32 };
+
+    let header = alphas[0]
+        + alphas[1].mul_base(Felt::from(transition_label))
+        + alphas[2].mul_base(addr_nxt - ONE)
+        + alphas[3].mul_base(ZERO);
+
+    let state = &main_trace.chiplet_hasher_state(i - 2)[4..];
+    let state_nxt = &main_trace.chiplet_hasher_state(i - 1)[4..];
+
+    header + build_value(&alphas[8..16], state_nxt) - build_value(&alphas[8..16], state)
+}
+
+/// Builds requests made to the hasher chiplet at the end of a block.
+fn build_end_block_request<E: FieldElement<BaseField = Felt>>(
+    main_trace: &MainTrace,
+    alphas: &[E],
+    i: usize,
+) -> E {
+    let op_label = RETURN_HASH_LABEL;
+    let addr = main_trace.addr(i) + Felt::from(7_u64);
+
+    let first_cycle_row = addr_to_row_index(addr) % 8 == 0;
+    let transition_label = if first_cycle_row { op_label + 16 } else { op_label + 32 };
+
+    let header =
+        alphas[0] + alphas[1].mul_base(Felt::from(transition_label)) + alphas[2].mul_base(addr);
+
+    let state = main_trace.decoder_hasher_state(i);
+    let digest = &state[..4];
+
+    header + build_value(&alphas[8..12], digest)
+}
+
+/// Builds requests made to the bitwise chiplet. This can be either a request for the computation
+/// of a `XOR` or an `AND` operation.
+fn build_bitwise_request<E: FieldElement<BaseField = Felt>>(
+    main_trace: &MainTrace,
+    is_xor: Felt,
+    alphas: &[E],
+    i: usize,
+) -> E {
+    let op_label = get_op_label(ONE, ZERO, is_xor, ZERO);
+    let a = main_trace.s0(i);
+    let b = main_trace.s1(i);
+    let z = main_trace.s0(i + 1);
+
+    alphas[0]
+        + alphas[1].mul_base(op_label)
+        + alphas[2].mul_base(b)
+        + alphas[3].mul_base(a)
+        + alphas[4].mul_base(z)
+}
+
+/// Builds `MLOADW` and `MSTOREW` requests made to the memory chiplet.
+fn build_mloadw_mstorew_request<E: FieldElement<BaseField = Felt>>(
+    main_trace: &MainTrace,
+    is_mstorew: bool,
+    alphas: &[E],
+    i: usize,
+) -> E {
```

Comment:
Instead of passing `is_mstorew` as boolean, we could probably just pass the label value here. Specifically, this could look like so:
```Rust
fn build_mloadw_mstorew_request<E: FieldElement<BaseField = Felt>>(
    main_trace: &MainTrace,
    op_label: Felt,
    alphas: &[E],
    i: usize,
) -> E {
```